include/ncengine/graphics/SkeletalAnimationTypes.h

```diff
@@ -14,13 +14,15 @@ namespace nc::graphics::anim
  */
 enum class Action : uint8_t
 {
-    Loop
+    Loop,
+    PlayOnce,
+    Stop
 };
 
 /**
  * @brief A simplified representation of a SkeletalAnimator state machine state sent to SkeletalAnimationSystem on state transition.
  */
-struct PlayState
+struct StateChange
```

Comment:
I think the three mesh paths here should be `std::string_view`s. Its a better default choice for event data where the receiver may likely only need to act upon the data before discarding it. They can still choose to make a copy if needed. Looking at the system, I think they're only being used to index maps anyway.